test/dotcom/alerts/subway_test.exs

```diff
@@ -0,0 +1,229 @@
+      # Setup
+      group_order = group_order()
+      alerts = []
+
+      # Exercise
+      grouped_alerts = group_alerts(alerts)
+
+      # Verify
+      Enum.each(group_order, fn group ->
+        assert Map.has_key?(grouped_alerts, group)
+      end)
+    end
+
+    test "alerts default to an empty list" do
+      # Setup
+      alerts = []
+
+      # Exercise
+      group_counts = group_alerts(alerts)
+
+      # Verify
+      assert Enum.all?(group_counts, fn {_group, alerts} -> Enum.empty?(alerts) end)
+    end
+
+    test "alerts get grouped by effect" do
+      # Setup
+      groups = groups()
+      random_group = Enum.random(groups) |> Kernel.elem(0)
+
+      effects =
+        Enum.find(groups, fn {group, _effects} -> group == random_group end) |> Kernel.elem(1)
+
+      random_effect = if Enum.empty?(effects), do: :foobarbaz, else: Enum.random(effects)
+
+      alert = Factories.Alerts.Alert.build(:alert, effect: random_effect)
+      alerts = [alert]
+
+      # Exercise
+      grouped_alerts = group_alerts(alerts)
+
+      # Verify
+      assert grouped_alerts |> Map.get(random_group) == alerts
+    end
+  end
+
+  describe "group_counts/1" do
+    test "all groups are present" do
+      # Setup
+      group_order = group_order()
+      alerts = []
+
+      # Exercise
+      grouped_alerts = group_alerts(alerts)
+
+      # Verify
+      Enum.each(group_order, fn group ->
+        assert Map.has_key?(grouped_alerts, group)
+      end)
+    end
+
+    test "counts default to 0" do
+      # Setup
+      alerts = []
+
+      # Exercise
+      group_counts = group_counts(alerts)
+
+      # Verify
+      assert Enum.all?(group_counts, fn {_group, count} -> count == 0 end)
+    end
+
+    test "alerts get counted by effect" do
+      # Setup
+      groups = groups()
+      random_group = Enum.random(groups) |> Kernel.elem(0)
+
+      effects =
+        Enum.find(groups, fn {group, _effects} -> group == random_group end) |> Kernel.elem(1)
+
+      random_effect = if Enum.empty?(effects), do: :foobarbaz, else: Enum.random(effects)
+
+      alert = Factories.Alerts.Alert.build(:alert, effect: random_effect)
+      alerts = [alert]
+
+      # Exercise
+      grouped_counts = group_counts(alerts)
+
+      # Verify
+      assert grouped_counts |> Map.get(random_group) == 1
+    end
+  end
+
+  describe "group_order/0" do
+    test "returns a list" do
+      # Exercise/Verify
+      assert group_order() |> is_list()
+    end
+  end
+
+  describe "groups/0" do
+    test "returns an ordered list" do
+      # Setup
+      group_order = group_order()
+
+      # Exercise
+      groups = groups()
+
+      # Verify
+      assert groups |> List.first() |> Kernel.elem(0) == Enum.at(group_order, 0)
+      assert groups |> List.last() |> Kernel.elem(0) == Enum.at(group_order, -1)
+    end
+  end
+
+  describe "find_group/1" do
+    test "returns 'Other' when the group is not found" do
+      # Setup
+      effect = :foobarbaz
```

Comment:
There's nothing special about `:foobarbaz`, right? Can it be replaced with a faker value?

Reply:
Talked offline and decided that moving forward, proper usage of Faker is no longer approval-blocking, thought it is still encouraged.